### DISCUSSION TITLE
Persist chats through sessions

### DISCUSSION THREAD

**Discussion by joaomendoncaa** [olimorris/codecompanion.nvim#520] [Q&A] [answered]
New here, absolutely insane work - thank you all.

Wondering if there's anyway we could persist chats through sessions. Maybe it's already implemented but I didn't figure it out glimpsing over the docs. The use case is simple, there's a lot of times you quit neovim to go do something else on the shell, and when you come back none of the previous chats are available even if you are in the same codebase.

**Comment by olimorris** [comment, marked as answer]
You're too kind and hope you're enjoying the plugin.

This is talked about in #139 and someone has implemented their own [save/restore functionality](https://github.com/olimorris/codecompanion.nvim/discussions/139#discussioncomment-11082730). The only reason I don't have it in the plugin is because the underlying structure of the chat buffer keeps changing. Perhaps in a few releases time when I've added in everything I want to it will be viable.
